### ISSUE TITLE
Crash on OTPublisher instance creation

### ISSUE THREAD

**Issue by TheSuperiorStanislav** [opentok/opentok-ios-sdk-samples-swift#193]
**Describe the bug**
App crashes with this message on OTPublisher

**To Reproduce**
Create a session, connect to a session, create OTPublisher, and disconnect from the session. Repeat multiple times until crashes. Sometimes it happened on the first attempt

**Expected behavior**
The app does not crash

**Device (please complete the following information):**
- sessionId, if applicable:
- iOS SDK version: 16.3
- OS and version: 16.3

**Additional context**
Stacktrace
```
Application Specific Information:
*** -[AVCaptureSession startRunning] startRunning may not be called between calls to beginConfiguration and commitConfiguration

Thread 18 Crashed:
0   CoreFoundation                  0x300a58d1c         <redacted>
1   libobjc.A.dylib                 0x32fb20ee4         objc_exception_throw
2   AVFCapture                      0x3476bc4f0         <redacted>
3   libdispatch.dylib               0x300407094         <redacted>
4   libdispatch.dylib               0x300408094         <redacted>
5   libdispatch.dylib               0x3003ec74c         <redacted>
6   libdispatch.dylib               0x3003ecdec         <redacted>
7   libsystem_pthread.dylib         0x3b7073dd4         _pthread_wqthread
8   libsystem_pthread.dylib         0x3b707393c         start_wqthread

Thread 0 name: Main thread
0   libsystem_kernel.dylib          0x375b92aac         mach_msg_trap
1   libsystem_kernel.dylib          0x375b9307c         mach_msg
2   CoreFoundation                  0x3009ccd88         <redacted>
3   CoreFoundation                  0x3009d1090         <redacted>
4   CoreFoundation                  0x3009e4250         CFRunLoopRunSpecific
5   AVFCore                         0x3125c66ac         <redacted>
6   AVFCapture                      0x3477305f4         <redacted>
7   AVFCapture                      0x3476acd44         <redacted>
8   WorkApp                         0x2052181e4         __52-[OTVideoCaptureIOSDefault setCaptureSessionPreset:]_block_invoke
9   libdispatch.dylib               0x300408094         <redacted>
10  libdispatch.dylib               0x3003ea464         <redacted>
11  WorkApp                         0x2052180b4         -[OTVideoCaptureIOSDefault setCaptureSessionPreset:]
12  WorkApp                         0x205212de0         -[OTPublisher initWithDelegate:settings:]
13  WorkApp                         0x2042485ec         OTPublisher.init
14  WorkApp                         0x204248590         OTPublisher.__allocating_init
```

**Comment by emilyvon** [comment]
I am also facing this issue after upgrading to v2.24.2 (previously running v2.21.0 without this problem).

The crash happens quite randomly but I managed to reproduce it using the sample app in this repo called "Basic-Video-Chat". (I have upgraded the pod version to be 2.24.2 and added extra logs to the sample app.)

Steps to reproduce (intermittently - I had done these steps for the whole day non stop and only happened twice):
1. Build app from Xcode to physical iPhone running 15.6.1
2. App trying to connect to the session for the first time after viewDidLoad()
3. App crashes at attempting to initialising an OTPublisher

![Screenshot 2023-03-24 at 3 53 28 pm](https://user-images.githubusercontent.com/12323442/227431775-c0058ea3-c60c-4fde-a82d-93918336b082.png)

![Screenshot 2023-03-24 at 3 53 31 pm](https://user-images.githubusercontent.com/12323442/227432132-3dd0ce2a-8bf5-47ae-b2ff-88396db45a5f.png)
